pkg/helm/controller/reconcile.go

```diff
@@ -94,38 +96,8 @@ func (r HelmOperatorReconciler) Reconcile(request reconcile.Request) (reconcile.
 	status := types.StatusFor(o)
 	log = log.WithValues("release", manager.ReleaseName())
 
-	deleted := o.GetDeletionTimestamp() != nil
-	pendingFinalizers := o.GetFinalizers()
-	if !deleted && !contains(pendingFinalizers, finalizer) {
-		log.V(1).Info("Adding finalizer", "finalizer", finalizer)
-		finalizers := append(pendingFinalizers, finalizer)
-		o.SetFinalizers(finalizers)
-		err = r.updateResource(o)
-
-		// Need to requeue because finalizer update does not change metadata.generation
-		return reconcile.Result{Requeue: true}, err
-	}
-
-	status.SetCondition(types.HelmAppCondition{
-		Type:   types.ConditionInitialized,
-		Status: types.StatusTrue,
-	})
-
-	if err := manager.Sync(context.TODO()); err != nil {
-		log.Error(err, "Failed to sync release")
-		status.SetCondition(types.HelmAppCondition{
-			Type:    types.ConditionIrreconcilable,
-			Status:  types.StatusTrue,
-			Reason:  types.ReasonReconcileError,
-			Message: err.Error(),
-		})
-		_ = r.updateResourceStatus(o, status)
-		return reconcile.Result{}, err
-	}
-	status.RemoveCondition(types.ConditionIrreconcilable)
```

Comment:
This section is moved to _after_ the deletion check and uninstall code. This is necessary to ensure that a CR can always be deleted. There _were_ cases where the CR had a deletion timestamp, but this call to `manager.Sync` was failing so we always errored out before getting to the `manager.Uninstall` call.